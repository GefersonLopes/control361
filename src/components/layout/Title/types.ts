export type TitlePageProps = {
  title: string;
  className?: string;
};
