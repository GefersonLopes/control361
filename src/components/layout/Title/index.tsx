import clsx from "clsx";

import type { TitlePageProps } from "./types";

function Title({ title, className }: TitlePageProps) {
  return (
    <h1
      className={clsx(
        "text-base font-semibold tracking-wide font-poppins color-light",
        className,
      )}
    >
      {title}
    </h1>
  );
}

export default Title;
