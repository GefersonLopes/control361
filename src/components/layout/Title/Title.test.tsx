import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import Title from "../Title";

describe("Title component", () => {
  it("renders an h1 element with the provided title text", () => {
    render(<Title title="My Page Title" />);
    const heading = screen.getByRole("heading", {
      level: 1,
      name: "My Page Title",
    });
    expect(heading).toBeInTheDocument();
  });

  it("applies the default classes", () => {
    render(<Title title="Default Classes" />);
    const heading = screen.getByText("Default Classes");
    expect(heading).toHaveClass(
      "text-base",
      "font-semibold",
      "tracking-wide",
      "font-poppins",
      "color-light",
    );
  });

  it("merges a custom className when provided", () => {
    render(
      <Title title="Custom Class" className="extra-class another-class" />,
    );
    const heading = screen.getByText("Custom Class");
    expect(heading).toHaveClass("extra-class", "another-class");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <Title title="Snapshot Title" className="snapshot-class" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
