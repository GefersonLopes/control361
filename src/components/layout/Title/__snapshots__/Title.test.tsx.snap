// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component matches snapshot 1`] = `
<DocumentFragment>
  <h1
    class="text-base font-semibold tracking-wide font-poppins color-light snapshot-class"
  >
    Snapshot Title
  </h1>
</DocumentFragment>
`;
